klippy/extras/virtual_sdcard.py

```diff
@@ -188,6 +253,11 @@ def work_handler(self, eventtime):
         logging.info("Exiting SD card print (position %d)", self.file_position)
         self.work_timer = None
         self.cmd_from_sd = False
+        if self.current_file is not None:
+            self.print_stats.note_pause()
+        else:
+            self.file_position = self.file_size = 0
+            self.print_stats.reset()
```

Comment:
I don't understand this change.  How could `self.currentfile == None` in this context?

Reply:
Yes, I should change that to use reset_file.